test/correctness/image_wrap.cpp

```diff
@@ -17,27 +17,59 @@ typedef map<string, vector<string>> CallGraphs;
 
 class CheckCalls : public IRVisitor {
 public:
+    CheckCalls(const map<string, Function> &funcs) : funcs(funcs) {}
+
     CallGraphs calls; // Caller -> vector of callees
-    string producer = "";
 private:
+    map<string, Function> funcs;
+    string producer = "";
+
     using IRVisitor::visit;
 
     void visit(const ProducerConsumer *op) {
-        string old_producer = producer;
-        producer = op->name;
-        calls[producer]; // Make sure each producer is allocated a slot
-        op->produce.accept(this);
-        producer = old_producer;
-
-        if (op->update.defined()) {
-            // Just lump all the update stages together
-            producer = op->name + ".update(" + std::to_string(0) + ")";
+        if (op->is_producer) {
+            assert(funcs.count(op->name));
+            Stmt produce = op->body;
+            Stmt update;
+
+            if (!funcs[op->name].updates().empty()) {
```

Comment:
Do we actually need to distinguish between calls from updates vs. producers in the test? It seems possibly excessive, and there's a lot of logic here to enable that...